packages/ui-form-field/src/FormFieldLayout/index.tsx

```diff
@@ -84,133 +79,122 @@ class FormFieldLayout extends Component<FormFieldLayoutProps> {
-        >
-          <FormFieldLabel
-            aria-hidden={this.elementType === 'fieldset' ? 'true' : undefined}
-          >
-            {this.props.label}
-          </FormFieldLabel>
-        </Grid.Col>
+        <span css={this.props.styles?.formFieldLabel}>{this.props.label}</span>
       )
-    } else if (this.elementType !== 'fieldset') {
-      // to avoid duplicate label/legend content
-      return this.props.label
-    } else {
-      return null
-    }
-  }
-
-  renderLegend() {
-    // note: the legend element must be the first child of a fieldset element for SR
-    // so we render it twice in that case (once for SR-only and one that is visible)
-    return (
-      <ScreenReaderContent as="legend">
-        {this.props.label}
-        {this.hasMessages && (
-          <FormFieldMessages messages={this.props.messages} />
-        )}
-      </ScreenReaderContent>
-    )
+    } else if (this.props.label) {
+      if (this.elementType == 'fieldset') {
+        return (
+          <legend id={this._labelId} style={{ display: 'contents' }}>
+            {this.props.label}
+          </legend>
+        )
+      }
+      // needs to be wrapped because it needs an `id`
+      return <div id={this._labelId}>{this.props.label}</div>
+    } else return null
   }
 
   renderVisibleMessages() {
     return this.hasMessages ? (
-      <Grid.Row>
-        <Grid.Col
-          offset={this.inlineContainerAndLabel ? undefined : 3}
-          textAlign={this.inlineContainerAndLabel ? 'end' : undefined}
-        >
-          <FormFieldMessages
-            id={this._messagesId}
-            messages={this.props.messages}
-          />
-        </Grid.Col>
-      </Grid.Row>
+      <FormFieldMessages
+        id={this._messagesId}
+        messages={this.props.messages}
+        gridArea="messages"
+      />
     ) : null
   }
 
   render() {
-    // any cast is needed to prevent Expression produces a union type that is too complex to represent errors
-    const ElementType = this.elementType as any
+    // Should be `<label>` if it's a FormField, fieldset if it's a group
+    const ElementType = this.elementType
 
     const { makeStyles, styles, messages, isGroup, ...props } = this.props
 
-    const { width, layout, children } = props
+    const { width, children } = props
 
-    const hasNewErrorMsg =
+    const hasNewErrorMsgAndIsGroup =
       !!messages?.find((m) => m.type === 'newError') && isGroup
+    // TODO use aria-errormessage here https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage
     return (
       <ElementType
-        {...omitProps(props, [
-          ...FormFieldLayout.allowedProps,
-          ...Grid.allowedProps
-        ])}
+        {...omitProps(props, [...FormFieldLayout.allowedProps])}
         css={styles?.formFieldLayout}
         style={{ width }}
-        aria-describedby={this.hasMessages ? this._messagesId : undefined}
```

Comment:
`aria-describedby` is no longer needed because SRs work fine if the `label`'s child is the control